modules/media/ajax/FileDownload.php

```diff
@@ -35,16 +36,34 @@
     ["file" => $file]
 );
 
-if (!file_exists($filePath)) {
-    error_log("ERROR: File $filePath does not exist");
-    header("HTTP/1.1 404 Not Found");
-    exit(5);
+// download from s3
+
+use Aws\S3\S3Client;
+$s3_download_status = false;
+
+$config      = \NDB_Config::singleton();
+$bucketName = $config->getSetting('AWS_S3_Default_Bucket');
```

Comment:
S3 urls are of the form `s3://bucketname/filename`. I think this setting is only for uploads, not downloads.